
# HG changeset patch
# User John H. Robinson, IV <jaqque@sbih.org>
# Date 1229206686 28800
# Node ID ceef975fe4e697423f61314b3fa5cf0d44d7a21c
# Parent 17b8b5652f973923c0e07baddeb5765d9acedf55
Do not allow a pop from array if no more items are left

Author: Tobias Stoeckmann
Retrieved from: http://paldium.homeunix.org/tobias/wmaker/
Submitted by: Gilbert Ashley <amigo@ibiblio.org>

--- a/WINGs/array.c	Sat Dec 13 14:17:09 2008 -0800
+++ b/WINGs/array.c	Sat Dec 13 14:18:06 2008 -0800
@@ -230,6 +230,8 @@
 void*
 WMPopFromArray(WMArray *array)
 {
+    wassertr(array->itemCount > 0);
+
     array->itemCount--;
 
     return array->items[array->itemCount];

